### PR TITLE
Handle out args either at beginning or end of the function sig.

### PR DESCRIPTION
**Pull request by ailzhang**

### CODE REVIEW

**Comment by JackCaoG** [comment]
I saw the breakage on the pytorch side related to `out`, did some pytorch pr changed the `out` signature?

**Comment by smessmer** [comment]
We're working on aligning the kernel signature with the schema as defined in native_functions.yaml, that means that kernels will get out arguments in the end of the argument list. It's a migration in progress though, we're not 100% finished yet.

**Review by JackCaoG — APPROVED** [review]
LGTM, do you need to add a torch_pin to this pr for testing?

File: scripts/gen.py [hunk 1]
```diff
@@ -795,33 +795,39 @@ def generate_aten_out(ctx, tree, rwxtree, fname, sig, rwsig, params, fnopts):
...
-    code += generate_outfn_result_copy(param_vars[0], tmp_result)
-    code += generate_exit_debug_code(tree, fname, param_vars[0], params,
+    code += generate_outfn_result_copy(out_vars[0], tmp_result)
+    code += generate_exit_debug_code(tree, fname, out_vars[0], params,
                                      param_vars)
-    code += '  return {};\n'.format(param_vars[0])
+    code += '  return {};\n'.format(out_vars[0])
   else:
-    for i in range(0, num_outputs):
+    for i in range(num_outputs):
```
Comment (davidel) [inline scripts/gen.py:822]: Along all our code we use to explicitly specify the base of the range, so I'd undo this change.

**Review by davidel — APPROVED** [review]
Minor nits.
